### ISSUE TITLE
Manager: track all changes made to a round and have it stored in state

### ISSUE THREAD

**Issue by michellema1208** [gitcoinco/grants-stack#1646]
**User Story:**
As a program manager
I want to be able to edit my round details
So that I can make necessary changes before the round starts

**Acceptance Criteria**
GIVEN that I am on my Round Settings tab AND the round voting period hasn't started yet
WHEN I click the Edit Round button
THEN all the round details components switch to their editable state
AND the Edit Round button is replaced by the Cancel and Update Round buttons

GIVEN that I am on the Round Settings tab in the editable state
WHEN I click Cancel
THEN all the round details components switch back to their uneditable state
AND the Cancel/Update Round buttons are replaced by the Edit Round button

GIVEN that I am on the round settings tab in the editable state AND have made changes to my settings
WHEN I click "Update Round"
THEN I see the Update Round modal
AND it tells me how many transactions I will need to sign to make the changes

GIVEN that I am looking at the Update Round modal
WHEN I click Cancel
THEN the modal closes and I return to the editable state

**Product & Design Links:**
[Figma](https://www.figma.com/file/guY5OKIgM18SNa62OysBoe/%5BWIP%5D-Linear-QF-Flow-v2?node-id=143-9624&t=oaaQsqSd6tVU7JGu-4)
[Loom of UX](https://www.loom.com/share/30521a647d074d49a7bc306809fbe450)

**Tech Details**

As a round operator, when I should be able to 
- enable edits on all the fields across the tabs  
- make changes to the form 

From the dapp side , the dapp should be able to 
- keep track of which all fields have been changed
- store them in the state

End Result: You should be able to console log the state and show which all fields have been updated

**Comment by bhargavaparoksham** [comment]
![Image](https://user-images.githubusercontent.com/22616592/236216770-5b44bccf-5206-4398-81eb-3e92e1a2f404.png)


We should allow users to edit the dates that are in the future. Can we add some user feedback in the UI so that users know that?

@michellema1208 @willsputra 

cc: @codenamejason

**Comment by bhargavaparoksham** [comment]
![Image](https://user-images.githubusercontent.com/22616592/236217910-41bc6d94-5990-4bb2-93d8-09663bb45c4a.png)

@michellema1208 I remember @nfrgosselin mentioning that round operator should not be allowed to reduce the matching pool amount. Is this still correct?

cc: @codenamejason  @thelostone-mc

**Comment by codenamejason** [comment]
<img width="692" alt="Screenshot 2023-05-04 at 9 35 04 AM" src="https://user-images.githubusercontent.com/9419140/236221208-4c75bf95-9e79-407b-b2e7-aeed5e41144c.png">

Is there a protocol level minimum donation? Should we prevent the round operator from putting less than say $1? 

@thelostone-mc @michellema1208

cc: @bhargavaparoksham

**Comment by michellema1208** [comment]
@codenamejason @bhargavaparoksham @KurtMerbeth - update on the above! 

1. We'll add some error states for when people try to set a date to a current or past date cc: @willsputra 
2. Even after a round has started, you should be able to edit your round end date. You cannot make the round end date the current date or past dates though. **This is ONLY for round end date - nothing else should be editable after the round starts.** 
3. Correct, you can't decrease the size of the matching pool. We'll also add an error state for if someone tries to decrease matching funds available. 
4. You can edit your eligibility criteria even after an application period has started. You can no longer edit eligibility criteria after the round starts. If we get feedback here, we can adjust down the road. 
5. There is no protocol level minimum donation. 

### To summarize

**You can't edit:** 
- Program Chain 
- Payout token 

**You can edit before the round start date:** 
- Round Name 
- Round Description 
- Support Info 
- Eligibility Requirements 
- App Start/End Dates (new dates must be in the future) 
- Round Start Date (new dates must be in the future) 
- Matching funds (can only increase) 
- Matching cap 
- Minimum Donation Threshold 
- Sybil Defense 

**You can edit before and after the round start date:**
- Round End Date
